src/main/java/org/reactome/release/downloaddirectory/GenerateGOAnnotationFile/CellularComponentAnnotationBuilder.java

```diff
@@ -2,88 +2,180 @@
     }
 
     /**
-     * Proteins reaching this part of the script are deemed valid for GO annotation. Retrieves the Cellular Compartment accession associated with this instance and then calls the GOA line generator.
-     * @param proteinInst -- GKInstance, Individual protein instance from the retrieved proteins.
-     * @param reactionInst -- GKInstance, Parent reaction that the protein is associated with.
-     * @param taxonIdentifier -- String, CrossReference ID of protein's species.
+     * Performs an AttributeQueryRequest on the incoming reaction instance. This will retrieve all protein's
+     * affiliated with the Reaction.
+     * @param reactionlikeEvent -- GKInstance from ReactionlikeEvent class.
+     * @return -- Set of GKInstances output from the AttributeQueryRequest.
      * @throws Exception -- MySQLAdaptor exception.
      */
-    private static String generateGOCellularCompartmentLine(GKInstance proteinInst, GKInstance reactionInst, String taxonIdentifier) throws Exception {
-        String reactomeIdentifier = REACTOME_IDENTIFIER_PREFIX  + GOAGeneratorUtilities.getStableIdentifierIdentifier(reactionInst);
-        GKInstance referenceEntityInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.referenceEntity);
-        String goCellularCompartmentAccession = getCellularCompartmentGOAccession(proteinInst);
-        String goaLine = "";
-        if (!goCellularCompartmentAccession.isEmpty()) {
-            goaLine = GOAGeneratorUtilities.generateGOALine(referenceEntityInst, CELLULAR_COMPONENT_LETTER, goCellularCompartmentAccession, reactomeIdentifier, TRACEABLE_AUTHOR_STATEMENT_CODE, taxonIdentifier);
-            GOAGeneratorUtilities.assignDateForGOALine(proteinInst, goaLine);
-        } else {
-            logger.info("Protein has no Cellular Compartment accession, skipping GO annotation");
+    static Set<GKInstance> retrieveProteins(GKInstance reactionlikeEvent) throws Exception {
+        // All EWAS' associated with each of the classes in the instructions will be output for this ReactionlikeEvent.
+        String[] outClasses = new String[]{ReactomeJavaConstants.EntityWithAccessionedSequence};
+        return InstanceUtilities.followInstanceAttributes(
+            reactionlikeEvent, getClassAttributeInstructionsToFollow(), outClasses
+        );
+    }
+
+    private static List<ClassAttributeFollowingInstruction> getClassAttributeInstructionsToFollow() {
+        List<ClassAttributeFollowingInstruction> classesToFollow = new ArrayList<>();
+        for (String className : getClassToClassAttributesMapForObtainingProteins().keySet()) {
+            List<String> classAttributes = getClassToClassAttributesMapForObtainingProteins().get(className);
+            classesToFollow.add(getClassAttributeFollowingInstruction(className, classAttributes));
+        }
+        return classesToFollow;
+    }
+
+    private static Map<String, List<String>> getClassToClassAttributesMapForObtainingProteins() {
```

Comment:
I think it would make more sense if `classToClassAttributesForFollowingInstructions` were a class member initialised in a constructor or static initialiser. This method will return the same thing, every single time - it shouldn't be a method.

Reply:
Okay, done